File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/quickstart/ProductClient.java
package io.micronaut.nats.docs.quickstart;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.nats.annotation.Subject;
import io.micronaut.nats.annotation.NatsClient;
// end::imports[]

@Requires(property = "spec.name", value = "QuickstartSpec")
// tag::clazz[]
@NatsClient // <1>
public interface ProductClient {

    @Subject("product") // <2>
    void send(byte[] data); // <3>
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/nats/docs/quickstart/ProductClient.py
from abc import ABC, abstractmethod

from micronaut.context.annotation import Requires
# tag::imports[]
from micronaut.nats.annotation import NatsClient, Subject
# end::imports[]


@Requires(property="spec.name", value="QuickstartSpec")
# tag::clazz[]
@NatsClient  # <1>
class ProductClient(ABC):

    @Subject("product")  # <2>
    @abstractmethod
    def send(self, data: bytes) -> None:  # <3>
        ...
# end::clazz[]
